core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -228,6 +260,40 @@ private[spark] class EventLoggingListener(
     fileSystem.rename(new Path(logPath + IN_PROGRESS), target)
   }
 
+  /**
+   * According to the updated event to modify the maintained event's metrics
+   * @param executorId  the executor whose metrics will be modified
+   */
+  private def updateModifiedMetrics(executorId: String): Unit = {
+    val toBeModifiedEvent = executorIdToModifiedMaxMetrics.get(executorId)
+    val latestEvent = executorIdToLatestMetrics.get(executorId)
+    if (toBeModifiedEvent.isEmpty) {
+      if (latestEvent.isDefined) executorIdToModifiedMaxMetrics.update(executorId, latestEvent.get)
+    } else {
+      val toBeModifiedMetrics = toBeModifiedEvent.get.executorMetrics.transportMetrics
+      if (toBeModifiedMetrics.isDefined) {
```

Comment:
`latestEvent` should not always be defined when `toBeModifiedEvent.isEmpty` is true. The case is that at the beginning, the `latestEvent` is empty before the first metrics update event received.

Thank you for you cleaner code style example, I've updated in my code.